modules/apps/29-fee/keeper/relay.go

```diff
@@ -0,0 +1,39 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/modules/core/exported"
+)
+
+// SendPacket wraps IBC ChannelKeeper's SendPacket function
+func (k Keeper) SendPacket(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI) error {
+	return k.ics4Wrapper.SendPacket(ctx, chanCap, packet)
+}
+
+// WriteAcknowledgement wraps IBC ChannelKeeper's WriteAcknowledgement function
+// ICS29 WriteAcknowledgement is used for asynchronous acknowledgements
+func (k Keeper) WriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI, acknowledgement []byte) error {
+	// retrieve the forward relayer that was stored in `onRecvPacket`
+	packetId := channeltypes.NewPacketId(packet.GetSourceChannel(), packet.GetSourcePort(), packet.GetSequence())
+	relayer, found := k.GetForwardRelayerAddress(ctx, packetId)
+	fmt.Print(relayer)
+	if !found {
+		return sdkerrors.Wrapf(types.ErrForwardRelayerAddressNotFound, "packet: %s, %s, %d", packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
+	}
```

Comment:
don't return an error here. Since counterparty address may not have been registered. Just return the ack with empty forward relayer address as you would in `OnRecvPacket`